packages/apps/plugins/plugin-space/src/components/PopoverRemoveObject.tsx

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useCallback, useRef } from 'react';
+
+import { Folder } from '@braneframe/types';
+import { LayoutAction, parseIntentPlugin, parseLayoutPlugin, useResolvePlugin } from '@dxos/app-framework';
+import { TypedObject, getSpaceForObject } from '@dxos/react-client/echo';
+import { Button, Popover, useTranslation } from '@dxos/react-ui';
+
+import { SPACE_PLUGIN } from '../meta';
+
+export const PopoverRemoveObject = ({ object, folder: propsFolder }: { object: TypedObject; folder: Folder }) => {
+  const { t } = useTranslation(SPACE_PLUGIN);
+  const deleteButton = useRef<HTMLButtonElement>(null);
+
+  const layoutPlugin = useResolvePlugin(parseLayoutPlugin);
+  const intentPlugin = useResolvePlugin(parseIntentPlugin);
+
+  const handleDelete = useCallback(async () => {
```

Comment:
@wittjosiah shouldn't all of this delete behavior, taking into account folders and such, be moved somewhere else? feels weird for it to just be here. feels like this component should just call it.

Reply:
good question, this is sort of a consequence of the confirm popover, otherwise the logic would be in the intent. Perhaps we should consider a way to inject a confirm step into the intent process